File: RandomPlayer.py
from LogicalBoard import *
from random import randint
import os

class RandomPlayer:

    def __init__(self, color, oponent_color, columns, rows, c, p):
        self.color = color
        self.oponent_color = oponent_color
        self.logicalBoard = LogicalBoard(columns, rows, c, p)

        if not os.path.isdir('log'):
            os.mkdir('log')
        self.log_file = open('log/random_player_' + self.color + '.log', 'a')

    def move(self, last_move=None):
        if last_move is not None:
            self.log("He moved: " + str(last_move))
            self.logicalBoard.makeMove(self.oponent_color, last_move)

        while True:
            move = randint(0, self.logicalBoard.columns - 1)
            if self.logicalBoard.isValidMove(move):
                break
        self.log("I moved: " + str(move))
        self.logicalBoard.makeMove(self.color, move)
        return move

    def log(self, msg):
        self.log_file.write(str(msg) + "\n")
        self.log_file.flush()

File: random_player.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <random>
#include <cstdlib>
#include <vector>
#include <stdio.h>
#include <sys/stat.h>

std::random_device rd;
std::mt19937 generator(rd());

void send(const std::string& msg) {
    std::cout << msg << std::endl;
}

void send(int msg) {
    std::cout << msg << std::endl;
}

int read_int() {
    std::string msg;
    std::cin >> msg;
    if (msg == "salir") {
        send("listo");
        std::exit(0);
    }
    return std::stoi(msg);
}

std::string read_str() {
    std::string msg;
    std::cin >> msg;
    if (msg == "salir") {
        send("listo");
        std::exit(0);
    }
    return msg;
}

int main() {

    //std::default_random_engine generator;
    std::string msg, color, oponent_color, go_first;
    int columns, rows, c, p, move;

    while (true) {
        color = read_str();
        oponent_color = read_str();

        columns = read_int();
        rows = read_int();
        c = read_int();
        p = read_int();

        std::uniform_int_distribution<int> do_move(0, columns - 1);
        std::vector<int> board(columns);

        for(int i=0; i<columns; ++i) board[i] = 0;

        go_first = read_str();
        if (go_first == "vos") {
            move = do_move(generator);
            board[move]++;
            send(move);
        }

        while (true) {
            msg = read_str();
            if (msg == "ganaste" || msg == "perdiste" || msg == "empataron") {
                break;
            }

            board[std::stoi(msg)]++;
            do {
                move = do_move(generator);
            } while(board[move] >= rows);
            
            board[move]++;
            send(move);
        }
    }

    return 0;
}